worker/src/handles/BackoffTimerHandle.cpp

```diff
@@ -0,0 +1,144 @@
+#define MS_CLASS "BackoffTimerHandle"
+// #define MS_LOG_DEV_LEVEL 3
+
+#include "handles/BackoffTimerHandle.hpp"
+#include "Logger.hpp"
+#include "MediaSoupErrors.hpp"
+#include <algorithm> // std::min()
+
+/* Instance methods. */
+
+BackoffTimerHandle::BackoffTimerHandle(
+  Listener* listener,
+  uint64_t baseTimeout,
+  BackoffAlgorithm backoffAlgorithm,
+  std::optional<uint64_t> maxBackoffTimeout,
+  std::optional<size_t> maxRestarts)
+  : listener(listener), baseTimeout(baseTimeout), backoffAlgorithm(backoffAlgorithm),
+    maxBackoffTimeout(maxBackoffTimeout), maxRestarts(maxRestarts)
+{
+	MS_TRACE();
+
+	if (this->baseTimeout > BackoffTimerHandle::MaxTimeout)
+	{
+		MS_THROW_ERROR(
+		  "base timeout (%" PRIu64 ") cannot be greater than %" PRIu64,
+		  this->baseTimeout,
+		  BackoffTimerHandle::MaxTimeout);
+	}
```

Comment:
What about replacing the block by calling `SetBaseTimeout()` which does the same?

Reply:
ok